packages/anasazi/epetra/test/OrthoManager/cxx_gentest.cpp

```diff
@@ -285,9 +285,10 @@ int main(int argc, char *argv[])
       MyOM->stream(Warnings) << "   ||<Y1,X1> - <X1,Y1>||     : " << err << endl;
       // test s.p.d. of <Y1,X1>: try to compute a cholesky
       lapack.POTRF('U',yTx1.numCols(),yTx1.values(),yTx1.stride(),&info);
-      TEUCHOS_TEST_FOR_EXCEPTION(info != 0,std::runtime_error,
-          "New <Y1,X1> not s.p.d.: couldn't computed Cholesky: info == " << info
-          << "\nyTx1: \n" << yTx1);
+      std::ostringstream msg("New <Y1,X1> not s.p.d.: couldn't computed Cholesky: info == ");
```

Comment:
Hm, I'd rather not build the `std::ostringstream` each time if we only use it when `info` is nonzero.  How about this?
```c++
if (info != 0) {
  // build the std::ostringstream
  yTx1.print (msg);
  TEUCHOS_TEST_FOR_EXCEPTION(true, std::runtime_error, msg); 
}
```

Reply:
I realize now that the above code is in a test, so I'm less worried about it.